src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -172,84 +206,172 @@ class instance_scarlet_monastery : public InstanceMapScript
                     return WhitemaneGUID;
                 case DATA_DOOR_WHITEMANE:
                     return DoorHighInquisitorGUID;
+                case DATA_DOOR_CHAPEL:
+                    return DoorChapelGUID;
+                default:
+                    return ObjectGuid::Empty;
+                    break;
             }
-
-            return ObjectGuid::Empty;
         }
 
         uint32 GetData(uint32 type) const override
         {
-            if (type == TYPE_MOGRAINE_AND_WHITE_EVENT)
-                return encounter;
-            else if (type == DATA_HORSEMAN_EVENT)
-                return encounter;
-            return 0;
+            switch (type)
+            {
+                case TYPE_MOGRAINE_AND_WHITE_EVENT:
+                    return encounter;
+                    break;
+                case DATA_HORSEMAN_EVENT:
+                    return encounter;
+                    break;
+                case TYPE_ASHBRINGER_EVENT:
+                    return ashencounter;
+                    break;
+                default:
+                    return 0;
+                    break;
+            }
         }
     private:
         ObjectGuid DoorHighInquisitorGUID;
+        ObjectGuid DoorChapelGUID;
         ObjectGuid MograineGUID;
         ObjectGuid WhitemaneGUID;
-        uint32 encounter;
+        uint32 encounter{};
+        uint32 ashencounter{};
+        GuidSet AshbringerNpcGUID;
     };
 };
 
+class at_scarlet_monastery_cathedral_entrance : public AreaTriggerScript
+{
+public:
+    at_scarlet_monastery_cathedral_entrance() : AreaTriggerScript("at_scarlet_monastery_cathedral_entrance") {}
+
+    bool OnTrigger(Player* player, AreaTrigger const* /*trigger*/) override
+    {
+        if (player->HasAura(AURA_OF_ASHBRINGER))
+        {
+            if (InstanceScript* instance = player->GetInstanceScript())
+            {
+                if (instance->GetData(TYPE_ASHBRINGER_EVENT) == NOT_STARTED)
+                {
+                    Creature* commanderMograine = ObjectAccessor::GetCreature(*player, instance->GetGuidData(DATA_MOGRAINE));
+                    if (commanderMograine && commanderMograine->IsAlive())
+                        commanderMograine->AI()->Talk(SAY_MOGRAINE_ASHBRBINGER_INTRO);
+                    instance->SetData(TYPE_ASHBRINGER_EVENT, IN_PROGRESS);
+                    return true;
+                }
```

Comment:
would like to see some continuity for the brackets after the if statements, on most you use them one one you don't. I would say don't for one liners

Reply:
This is still valid

Reply:
Do you mean all "if" needs to be followed by curly braces? And a blank line?

Reply:
Such?

Reply:
I mean one line does not need it

```cpp
if ()
    if ()
        if ()
        {
           line1
           line2
        }

Reply:
Yours now look like this

```cpp
if ()
{
    if ()
{
        if ()
        {   
           line1
           line2

           if ()
               line1
        }